aten/src/ATen/native/mps/operations/TensorCompare.mm

```diff
@@ -459,11 +464,19 @@ static void where_kernel_mps(TensorIterator& iter) {
         Placeholder(cachedGraph->selfTensor_, self, /*mpsShape=*/nullptr, /*gatherTensorData=*/true, selfDataType);
     Placeholder otherPlaceholder =
         Placeholder(cachedGraph->otherTensor_, other, /*mpsShape=*/nullptr, /*gatherTensorData=*/true, otherDataType);
-    Placeholder outputPlaceholder = Placeholder(cachedGraph->outputTensor_, out);
+    Placeholder outputPlaceholder = Placeholder(cachedGraph->outputTensor_,
+                                                needsGather(out) ? out_ : out,
+                                                /*mpsShape=*/nullptr,
+                                                /*gatherTensorData=*/needsGather(out),
+                                                getMPSScalarType(out.scalar_type()));
 
     auto feeds = dictionaryFromPlaceholders(conditionPlaceholder, selfPlaceholder, otherPlaceholder);
     runMPSGraph(stream, cachedGraph->graph(), feeds, outputPlaceholder);
   }
+
+  if (needsGather(out)) {
+    out.copy_(out_);
```

Comment:
Does this intend to keep the strides / layout the same? If not, you can request copy operations to be continuous with contiguous_memory_format.

Reply:
We do want to keep out's strides/layout. Did you suggest otherwise?

Reply:
I see, nto sure there is another way of doing this then if MPS graph doesn't support non-contiguous inputs